user-service/src/controllers/userController.ts

```diff
@@ -0,0 +1,56 @@
+import { Request, Response } from "express";
+import bcrypt from "bcrypt";
+import User from '../models/userModel';
+import generateTokenAndSetCookie from '../lib/generateToken';
+import mongoose from "mongoose";
+
+export async function createUser(req: Request, res: Response) {
+  try {
+    const { username, email, password } = req.body;
+
+    // sanity check
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+		if (!emailRegex.test(email)) {
+			return res.status(400).json({ error: "Invalid email format" });
+		}
+
+    const existingUser = await User.findOne({ username });
+		if (existingUser) {
+			return res.status(400).json({ error: "Username is already taken" });
+		}
+
+		const existingEmail = await User.findOne({ email });
+		if (existingEmail) {
+			return res.status(400).json({ error: "Email is already taken" });
+		}
+
+		if (password.length < 10) {
```

Comment:
maybe can reduce it so that we can easily put short passwords for testing purposes 🥲

Reply:
i mean does it truly matter when the frontend password validation is even stricter :')

Reply:
Ok

Reply:
we have frontend password validation?

Reply:
why wd i want to write `admin12345` why i can write `asdf` :) all for efficiency

Reply:
![IMG_6206](https://github.com/user-attachments/assets/f016491e-21de-4745-862b-37f8cadbb300)

oof ow owie

Reply:
oof this is my first time seeing the login page ngl